### ISSUE TITLE
i18n to translation  multi word not work

### ISSUE THREAD

**Issue by ScorpioRen** [gogf/gf#2370]
my code runing in linux amd64 use golang 1.19.3 and gf version v2.2.5 use i18n to translation 
1) zh-CN/vluc.toml
  "vluc.request"  ="请求"
   "vluc.forbidden"="被禁止的"
2) test
   i18n := gi18n.New()
  actx := gi18n.WithLanguage(context.TODO(), "zh-CN")
  fmt.Println(i18n.Translate(actx, `{#vluc.request}{#vluc.request}`))
3）output
   {#vluc.request}{#vluc.request} is not to translate to zh-CN but if use  fmt.Println(i18n.Translate(actx, `vluc.request`)) can translate 
4)  it's bug?

**Comment by ScorpioRen** [comment]
I found that it was probably the RE matching gi18n_manager.go change pattern \{#(\w+)\} to  \{#([\w.]+)} is ok

**Comment by gqcn** [comment]
@ScorpioRen Yes, the variable name in i18n is designed as `\w+`, which does not support character `.`.

**Comment by ScorpioRen** [comment]
I'm find the gvalid i18n file content like "gf.gvalid.rule.date" ,so I thought it was okay.thanks

**Comment by lingcoder** [comment]
你用的 v2.2.5 上 {#vluc.request} 不翻是对的，那会儿占位符按 \w+ 匹配，不认中间的点。gqcn 当时也说明了这点。   后来 #2652 改成了 (.+?)，v2.4.3 起 {#a.b} 这种可以。{#a}{#b} 拼在一起现在单测也有覆盖。  直接 Translate(ctx, "vluc.request") 是整 key 查找，和 {#...} 占位不是一条路。 升到 ≥ v2.4.3 即可，这单按已修复关了。
